src/rest/adapter.js

```diff
@@ -280,12 +280,13 @@ export default class RestAdapter extends Adapter {
         } else {
           data = {};
         }
-        
+
         serializerOptions = defaults(serializerOptions, {context: context, xhr: xhr});
-        
+
         // TODO: handle other errors codes such as 409
         // determine serializer behavior off of xhr response code
-        if(xhr.status === 422) {
+        let {error, ...rest} = data;
```

Comment:
Note that I'm only checking for `data.error`, which should be safe. I could also check for `data.errors` but I'd have to fix a lot of tests. And I think the arbitrary 422 errors that we send down all look like `error: {message: '', type: ''}` like so:

![screen shot 2016-12-05 at 3 11 59 pm](https://cloud.githubusercontent.com/assets/148534/20906556/850a011a-bafd-11e6-8e1e-ebeb24e6e41f.png)